broadinstitute/single_cell_portal_core
# Simplify logic for "Show all" / "Hide all" in scatter plot legend (SCP-3890)

This refactors code for the "Show all" and "Hide all" links atop the custom scatter plot legend.  

Previously, whether each link should be enabled was encoded as React state variables.  As Devon [noted](https://github.com/broadinstitute/single_cell_portal_core/pull/1254#pullrequestreview-806299496), that's unnecessary.  Now those variables are simply computed without encoding their values as React state.

This helps maintainability by condensing code from two modules into one more coherent module, and by clearer variable naming.

To test:
* Open study that has visualizations
* Note that all legend entries are shown, with "Show all" disabled and "Hide all" enabled
* Deselect a legend entry
* Verify that "Show all" and "Hide all" are both enabled
* Click "Hide all"
* Verify that no legend entries are shown
* Click "Show all"
* Verify that all legend entries are shown

This satisfies SCP-3890.
